x/valset-pref/simulation/sim_msgs.go

```diff
@@ -31,88 +31,123 @@ func RandomMsgSetValSetPreference(k valsetkeeper.Keeper, sim *osmosimtypes.SimCt
 func RandomMsgDelegateToValSet(k valsetkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*types.MsgDelegateToValidatorSet, error) {
 	delegator := sim.RandomSimAccount()
 	// check if the delegator valset created
-	err := GetRandomExistingValSet(ctx, k, sim, delegator.Address)
+	_, err := GetRandomDelegations(ctx, k, sim, delegator.Address)
 	if err != nil {
 		return nil, err
 	}
 
-	delegationCoin := sim.RandExponentialCoin(ctx, delegator.Address)
+	amount := sim.BankKeeper().GetBalance(ctx, delegator.Address, sdk.DefaultBondDenom).Amount
+	if !amount.IsPositive() {
+		return nil, fmt.Errorf("balance is negative")
+	}
```

Comment:
Why are we getting negative balances? That shouldn't be possible. My guess is, if the DefaultBondDenom is not present in the address it returns -1 which means not present. If that is the case, please update the error message.

Reply:
yup just updated